packages/components/src/components/common/table/hooks/hooks.tsx

```diff
@@ -209,10 +212,10 @@ export const useTableKeyHandler = <TData,>({ table, data, options }: TableKeyboa
 };
 
 const toggleExpand = <TData,>(row: Row<TData>, expand: boolean, loadChildren?: (row: Row<TData>) => void) => {
-  if (expand && !row.getIsExpanded()) {
+  if (expand && row && row.getCanExpand() && !row.getIsExpanded()) {
```

Comment:
mark the row as optional so it is clear
```suggestion
const toggleExpand = <TData,>(expand: boolean, row?: Row<TData>, loadChildren?: (row: Row<TData>) => void) => {
  if (expand && !row.getIsExpanded()) {
  if (expand && row && row.getCanExpand() && !row.getIsExpanded()) {
```